File: src/19.ts
import fs from 'fs'

const source = fs.readFileSync('./res/popular-names.txt', 'utf-8')
console.log(
    [
        ...source
            .split('\n')
            .map(row => row.split('\t')[0])
            .reduce((countMap: Map<string, number>, name) => {
                countMap.set(name, (countMap.get(name) || 0) + 1)
                return countMap
            }, new Map())
            .entries()
    ]
        .sort(([nameA, countA], [nameB, countB]) => countB - countA)
)

File: src/19.py
# cut -f 1 res/popular-names.txt | sort | uniq -c | sort -nr

from helper.file_helper import read_string

source = read_string("./res/popular-names.txt")
names = tuple(map(lambda row: row.split("\t")[0], source.split("\n")))
uniq_names = frozenset(names)
print("\n".join(sorted(uniq_names, key=lambda name: names.count(name), reverse=True)))
